### ISSUE TITLE
[RFE] image-based gadgets: support stack map

### ISSUE THREAD

**Issue by alban** [inspektor-gadget/inspektor-gadget#2553]
## Current situation

While eBPF programs in image-based gadgets can use BPF_MAP_TYPE_STACK_TRACE maps, the content of those stack maps cannot be exposed to userspace in an event.

In image-based gadgets, ring buffer maps are typed: the BTF and the gadget metadata specify the struct and its list of fields. But it is self-contained and cannot currently reference an entry in a stack map.

## Impact

* The built-in gadget profile-cpu cannot be implemented as an image-based gadget.
* Image-based gadgets cannot meaningfully use BPF_MAP_TYPE_STACK_TRACE maps.

## Ideal future situation

References to entries of stack maps can be done from:
* The type of events send in a ring buffer maps
* The type a field in a hash map

## Implementation options

include/gadget/types.h has a new type:
```
struct gadget_stack_ref {
    int stack_map; // fd of the stack map
    int stack_id; // value returned by bpf_get_stackid()
}
```

So userspace can interpret a field of type `gadget_stack_id` as an index in a stack map.
The stack_map fd can be initialized with relocations in the same way the argument to bpf_map_lookup_elem is relocated to a map fd.

## Additional information

* An image-based version of the profile-cpu gadget would need to support stack map: #1925
* This is part of the epic #1929

This issue is about using stack maps from image-based gadgets.
Unwinding user stack will be for a future issue.

**Comment by i-Pear** [comment]
@alban You could assign this to me. I think it would be better for me to design the framework once I become more acquainted with the project. I will implement this after completing the basic uprobe tasks, if it's not urgent.

**Comment by mauriciovasquezbernal** [comment]
Fixed by #2671
